### PR TITLE
chore(docs): record memory/perf insights in Run_semgrep

### PR DESCRIPTION
**Pull request by emjin**
Since I'm closing off the memory investigation for now, I don't want to lose
the insights we gained. I'm recording them for now in a README in runner, as
well as some other notes that might be useful. This is not meant to be
exhaustive.

PR checklist:

- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Changelog guidelines](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry)
- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by kopecs — APPROVED** [review]

**Comment by aryx** [comment]
what about putting it in semgrep-core/doc/?
Better chance to be found I think.

File: semgrep-core/src/runner/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Notes
```
Comment (IagoAbal) [inline semgrep-core/src/runner/README.md]: I think this is not that long that it could go in `Run_semgrep.mli`.

Comment (IagoAbal) [reply semgrep-core/src/runner/README.md]: Also we had a document (maybe in Quip) with the "current perf issues", we could have one in Notion too, and maybe it's a better place.

Comment (emjin) [reply semgrep-core/src/runner/README.md]: I ended up putting it in the mli file; I like having it close to the code. Not sure I'd think to check Notion or keep it up to date

**Review by IagoAbal — APPROVED** [review]

**Pull request merged**